src/Compilers/CSharp/Portable/Symbols/SymbolExtensions.cs

```diff
@@ -839,5 +839,12 @@ internal static bool HasAsyncMethodBuilderAttribute(this Symbol symbol, [NotNull
             builderArgument = null;
             return false;
         }
+
+        internal static bool IsRequired(this Symbol symbol)
+            => symbol switch
+            {
+                FieldSymbol { IsRequired: true } or PropertySymbol { IsRequired: true } => true,
```

Comment:
Consider using something like `symbol is FieldSymbol { IsRequired: true } or PropertySymbol { IsRequired: true }` here